clang/lib/Driver/ToolChains/AIX.cpp

```diff
@@ -259,6 +260,61 @@ void aix::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   // Specify linker input file(s).
   AddLinkerInputs(ToolChain, Inputs, Args, CmdArgs, JA);
 
+  // Add sanitizer libraries.
+  const SanitizerArgs &Sanitize = ToolChain.getSanitizerArgs(Args);
+  const char *sanitizer = nullptr;
+  bool NeedsSanitizerDeps = false;
+  // For now, only support address sanitizer.
+  if (Sanitize.needsAsanRt())
+    sanitizer = "AddressSanitizer";
+
+  if (sanitizer) {
+    if (Sanitize.needsSharedRt()) {
+      ToolChain.getDriver().Diag(diag::err_drv_unsupported_sanitizer)
+          << "shared" << sanitizer << "AIX";
+      return;
+    }
+    NeedsSanitizerDeps = addSanitizerRuntimes(ToolChain, Args, CmdArgs);
+  }
+
+  // Add sanitizer runtime dependencies.
+  // Note: having the static runtime linked into shared libraries can
+  // lead to multiple copies of the runtime with AIX's linkage model,
+  // so disallow that.
+  if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles,
+                   options::OPT_shared, options::OPT_r)) {
+    if (NeedsSanitizerDeps)
+      linkSanitizerRuntimeDeps(ToolChain, Args, CmdArgs);
+  }
+
+  // We won't add the static sanitizer libraries to the DSO, but we will
+  // introduce the undefined sanitizer symbols like __asan_init to the DSO. On
+  // AIX, this undefined sanitizer symbol cannot pass final link. Add the
+  // import file to make these undefined symbols be resolved at runtime.
+  if (Args.hasArg(options::OPT_shared) &&
+      ToolChain.getSanitizerArgs(Args).needsAsanRt()) {
+    SmallString<128> SanRTSymbolList;
+    (Twine(ToolChain.getRuntimePath().value_or(".")) +
+     "/asan.link_with_main_exec.txt")
+        .toVector(SanRTSymbolList);
+    if (llvm::sys::fs::exists(SanRTSymbolList))
+      CmdArgs.push_back(Args.MakeArgString(Twine("-bI:") + SanRTSymbolList));
+    else
+      ToolChain.getDriver().Diag(diag::err_drv_missing_sanitizer_file)
+          << sanitizer << "import";
+    if (ToolChain.getSanitizerArgs(Args).linkCXXRuntimes()) {
+      SanRTSymbolList.clear();
+      (Twine(ToolChain.getRuntimePath().value_or(".")) +
+       "/asan_cxx.link_with_main_exec.txt")
+          .toVector(SanRTSymbolList);
+      if (llvm::sys::fs::exists(SanRTSymbolList))
+        CmdArgs.push_back(Args.MakeArgString(Twine("-bI:") + SanRTSymbolList));
+      else
+        ToolChain.getDriver().Diag(diag::err_drv_missing_sanitizer_file)
+            << sanitizer << "c++ import";
```

Comment:
Minor nit: Use capital "C" for "C++".
```suggestion
            << sanitizer << "C++ import";
```